File: ndannamo_project/backend/src/main/java/com/example/backend/model/OvernightStay.java
package com.example.backend.model;


import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;
import java.util.Set;

@Entity
@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@DiscriminatorValue("OVERNIGHT_STAY")
public class OvernightStay extends AttachableEntity {

    @OneToMany(mappedBy = "overnightStay", cascade = CascadeType.ALL, fetch = FetchType.EAGER, orphanRemoval = true)
    private List<Night> travelDays;

    @NotNull
    private LocalDate startDate;
    @NotNull
    private LocalDate endDate;

    private LocalTime startCheckInTime;
    private LocalTime endCheckInTime;
    private LocalTime startCheckOutTime;
    private LocalTime endCheckOutTime;


    private String address;
    private String contact;
    private String name;



}

File: ndannamo_project/frontend/src/models/OvernightStay.js
export default class OvernightStay {
    constructor(id, name, startDate, endDate, startCheckInTime, endCheckInTime, startCheckOutTime, endCheckOutTime, address, contact) {
        this.id = id;
        this.name = name;
        this.startDate = startDate;
        this.endDate = endDate;
        this.startCheckInTime = startCheckInTime;
        this.endCheckInTime = endCheckInTime;
        this.startCheckOutTime = startCheckOutTime;
        this.endCheckOutTime = endCheckOutTime;
        this.address = address;
        this.contact = contact;
    }
}


